crates/handlers/src/compat/login.rs

```diff
@@ -565,9 +579,13 @@ async fn user_password_login(
         .user()
         .find_by_username(username)
         .await?
-        .filter(mas_data_model::User::is_valid)
+        .filter(|user| user.deactivated_at.is_none())
         .ok_or(RouteError::UserNotFound)?;
 
+    if user.locked_at.is_some() {
+        return Err(RouteError::UserLocked);
```

Comment:
This is quite easy to test, could you add a test for this case?

Reply:
d31af23e